scrape/scrape.go

```diff
@@ -735,11 +755,15 @@ func (s *targetScraper) scrape(ctx context.Context, w io.Writer) (string, error)
 		}
 	}
 
-	_, err = io.Copy(w, s.gzipr)
+	n, err := io.Copy(w, io.LimitReader(s.gzipr, s.bodySizeLimit))
```

Comment:
Do we want to limit the size uncompressed or compressed?

Reply:
uncompressed. 
1. beacuse we want protect server OOM，server will decompress when read body，use uncompressed  size memory . 
2. if we limit the compressed body，it could get a unexpected EOF error when read and decompress incomplete gzip bytes. 

i should add some comment at config field.